### ISSUE TITLE
[Bug] `beartype.claw.beartype_this_package()` + `async` = :scream: 

### ISSUE THREAD

**Issue by msvensson222** [beartype/beartype#462]
Hey! Really appreciate the beartype package - using it a lot!

So far not many issues, but I ran across an interesting behaviour when trying to use it in combination with async functions, and `asyncio`.  (Everything below is Python 3.11 and beartype 0.19.0)

Assume I have the following folder structure;
```
├── main.py
├── my_package
│   ├── __init__.py
│   ├── __pycache__
│   └── my_functions
```
with my `__init__.py` file:
```python
from beartype.claw import beartype_this_package

beartype_this_package()
```

and `my_functions` (note that they are all incorrectly type-hinted):
```python
from beartype import beartype


def my_normal_fun() -> int:
    return "Hello"


async def my_async_fun() -> int:
    return "Hello async"


@beartype
async def my_async_fun_with_explicit_beartype_decorator() -> int:
    return "Hello async"

```

and `main.py`:
```python
import asyncio

from beartype.roar import BeartypeCallHintReturnViolation

from my_package.my_functions.my_functions import (
    my_async_fun,
    my_async_fun_with_explicit_beartype_decorator,
    my_normal_fun,
)

if __name__ == "__main__":
    try:
        asyncio.run(my_async_fun())
        print("Beartype did not work when calling my_async_fun")
    except BeartypeCallHintReturnViolation:
        print("Beartype worked!")

    try:
        asyncio.run(my_async_fun_with_explicit_beartype_decorator())
        print(
            "Beartype did not work when calling my_async_fun_with_explicit_beartype_decorator"
        )
    except BeartypeCallHintReturnViolation:
        print("Beartype worked!")

    try:
        my_normal_fun()
        print("Beartype did not work when calling my_normal_fun")
    except BeartypeCallHintReturnViolation:
        print("Beartype worked!")

```
I'm expecting `beartype_this_package()` to be applied to all methods, hence catch all type issues, but the output of running the `main.py` file is:
```bash
Beartype did not work when calling my_async_fun
Beartype worked!
Beartype worked!
```

Am I doing something wrong, or simply does beartype_this_package() not work for async/asyncio ?

**Comment by leycec** [comment]
**Guh!** Thanks so much for catching this, @msvensson222. You're doing everything right. Your dark suspicions are entirely correct. It's `beartype_this_package()` that is falling down here. It looks like we conveniently "forgot" to unit test `beartype.claw` import hooks against `async` callables. So sorry about that! This is what it feels like when bears cry.

![](https://media1.tenor.com/m/vAHlOsfhWEcAAAAd/grizzly-we-bare-bears.gif)
<sup>*@beartype 0.19.0 weeps for your codebase*</sup>

Thankfully, this is a trivial bug. We'll get all this patched up for our upcoming @beartype 0.19.1 patch release. Oh – and have a great Canadian thanksgiving. May we all eat turkeys until we're stuffed to the gills. :turkey: :yum:

**Comment by msvensson222** [comment]
Cool, thanks for the reponse! And even more thanks for maintaining such an awesome library!

**Comment by leycec** [comment]
**Resolved by ae8e66686dbb41a.** Thanks so much for your threnodic patience, @msvensson222. All `beartype.claw` import hooks now transparently type-check coroutines and asynchronous generators – exactly as AI ML QA demands. <sup>so many acronyms :sob:</sup>

We'll be dropping this with our upcoming `beartype 0.20.0rc0` (i.e., first release candidate) release in a week or two. I'll ping you here on GitHub when it lands! Until then, may the progressive Stockholm death metal be with you always.

![](https://media1.tenor.com/m/QvYdzwACui4AAAAd/amon-amarth.gif)
<sup>*channelling hard `beartype 0.20.0rc0` vibes here*</sup>
